build-tools/packages/build-tools/src/common/monoRepo.ts

```diff
@@ -33,26 +34,85 @@ function sentenceCase(str: string) {
     public readonly version: string;
+
+    /**
+     * Creates a new monorepo.
+     *
+     * @param kind The 'kind' of monorepo this object represents.
+     * @param repoPath The path on the filesystem to the monorepo. This location is expected to have either a
+     * package.json file with a workspaces field, or a lerna.json file with a packages field.
+     * @param ignoredDirs Paths to ignore when loading the monorepo.
+     */
     constructor(public readonly kind: MonoRepoKind, public readonly repoPath: string, ignoredDirs?: string[]) {
+        this.version = "";
         const lernaPath = path.join(repoPath, "lerna.json");
-        if (!existsSync(lernaPath)) {
-            throw new Error(`ERROR: lerna.json not found in ${repoPath}`);
+        const packagePath = path.join(repoPath, "package.json");
+        let versionFromLerna = false;
+
+        if (existsSync(lernaPath)) {
+            const lerna = readJsonSync(lernaPath);
+            if (lerna.version !== undefined) {
+                console.log(`${kind}: Loading version (${lerna.version}) from ${lernaPath}`);
+                this.version = lerna.version;
+                versionFromLerna = true;
+            }
+
+            if (lerna.packages !== undefined) {
+                console.log(`${kind}: Loading packages from ${lernaPath}`);
+                for (const dir of lerna.packages as string[]) {
+                    // TODO: other glob pattern?
+                    const loadDir = dir.endsWith("/**") ? dir.substr(0, dir.length - 3) : dir;
+                    this.packages.push(...Packages.loadDir(path.join(this.repoPath, loadDir), MonoRepoKind[kind], ignoredDirs, this));
+                }
+                return;
+            }
+        }
+
+        if (!existsSync(packagePath)) {
+            throw new Error(`ERROR: package.json not found in ${repoPath}`);
         }
-        const lerna = readJsonSync(lernaPath);
-        for (const dir of lerna.packages as string[]) {
-            // TODO: other glob pattern?
-            const loadDir = dir.endsWith("/**") ? dir.substr(0, dir.length - 3) : dir;
-            this.packages.push(...Packages.loadDir(path.join(this.repoPath, loadDir), MonoRepoKind[kind], ignoredDirs, this));
+        const pkgJson = readJsonSync(packagePath);
+        if (pkgJson.version === undefined && !versionFromLerna) {
+            this.version = pkgJson.version;
+            console.log(`${kind}: Loading version (${pkgJson.version}) from ${packagePath}`);
+        }
+
+        if (pkgJson.workspaces !== undefined) {
+            console.log(`${kind}: Loading packages from ${packagePath}`);
+            for (const dir of pkgJson.workspaces as string[]) {
+                this.packages.push(...Packages.loadGlob(dir, kind, ignoredDirs, this));
+            }
+            return;
         }
-        this.version = lerna.version;
+        fatal(`Couldn't find lerna.json or package.json, or they were missing expected properties.`);
```

Comment:
Nit: Might be nice to list the missing properties here to make the feedback more actionable if this is hit.

Reply:
Yeah, it's more bookkeeping than I'd like to do at this point.